Mathlib/Combinatorics/SimpleGraph/Diam.lean

```diff
@@ -228,4 +285,78 @@ lemma connected_iff_diam_ne_zero [Fintype α] [Nontrivial α] : G.Connected ↔
 
 end diam
 
+section radius
+
+/-- The radius of a simple graph is the minimum eccentricity of any vertex. -/
+noncomputable def radius (G : SimpleGraph α) : ℕ∞ :=
+  ⨅ u, G.eccent u
+
+/-- The center of a simple graph is the set of vertices with eccentricity equal to the radius. -/
+def center (G : SimpleGraph α) : Set α :=
+  {u | G.eccent u = G.radius}
+
+lemma center_nonempty [Nonempty α] : G.center.Nonempty :=
+  ENat.exists_eq_iInf G.eccent
+
+lemma radius_eq_iInf_iSup_edist : G.radius = ⨅ u, ⨆ v, G.edist u v :=
+  rfl
+
+lemma radius_le_eccent (u : α) : G.radius ≤ G.eccent u :=
+  iInf_le G.eccent u
+
+lemma radius_eq_top_of_not_connected (h : ¬G.Connected) :
+    G.radius = ⊤ := by
+  simp [radius, eccent_eq_top_of_not_connected h]
+
+lemma radius_eq_top_of_isEmpty [IsEmpty α] : G.radius = ⊤ :=
```

Comment:
Is there a general G.radius = top iff lemma around here? Or maybe one under the assumption that alpha is nontrivial or something, which mentions connectivity?

Reply:
The one you describe is more general, but I think the practical one assumes both those instances, so let's do that one